fix: 🎨 use cache.teia.rocks

melMass opened the pull request:
Draft to identify potential issues/needs.

melMass committed:
fix: 🎨 more consistent use of toURL methods

melMass committed:
feat: ⚡️ add typing

melMass committed:
fix: ⚡️ add cache.teia.rocks

melMass committed:
fix: 🎨 use the CDN for SUBJKT logo/icon

melMass committed:
Use the CDN everywhere (test)

melMass committed:
fix: ⚡️ use raw for GLTF

melMass committed:
fix: ⚡️ use raw for audio

melMass committed:
fix: ⚡️ use raw for pdfs

melMass committed:
fix: ⚡️ fallback to video for gifs

melMass committed:
fix: ⚡️ try to catch different errors

melMass committed:
fix: 🎨 catch pdf loading errors

melMass committed:
fix: ⚡️ use markdown on the feedback component

melMass committed:
fix: 🎨 load pdfs from IPFS gateway

melMass committed:
fix: ⚡️ better type detection

use response headers prior, not on error

melMass committed:
fix: ⚡️ nahh use the fallback and only for Gifs

melMass committed:
fix: ⚡️ don't display burned tokens on the friends feed

Closes #119

melMass committed:
fix: ⚡️ typedef and use 'nft' object (wip)

melMass committed:
feat: 🎨 use api.teia.rocks endpoint

melMass committed:
fix: ⚡️ continue

melMass committed:
fix: 🎨 use margin instead of a div

melMass committed:
fix: ⚡️ continue, move transfer to tabs

melMass committed:
fix: ⚡️ fix non collab transfers

melMass committed:
fix: 🎨 better transfer feedback

melMass committed:
fix: ⚡️ "better" logic for the block_list

The full ban list was passed to EACH objkts 🥹
This solves that by relying on the context

melMass committed:
fix: 🎨 add new metadata in infos

Closes #183

melMass committed:
fix: ⚡️ custom rights_uri

melMass committed:
fix: ⚡️ block_list for listings

melMass committed:
Merge branch 'main' into mel/feat/tc-thumnails

melMass committed:
fix: ⚡️ typing

melMass committed:
Merge remote-tracking branch 'origin/develop' into mel/feat/tc-thumnails

melMass committed:
fix: ⚡️ post merge

melMass committed:
fix: 🔥 use use-settings for bans

melMass committed:
fix: ⚡️ from merge

nft param

melMass committed:
fix: 🔥 different size logic

- display_uri (used in feeds, and preloading of some formats) - medium
- artifact_uri (used in detail view) - raw

xat commented on src/components/item-info/index.js:
```diff
@@ -22,13 +23,18 @@ export const ItemInfo = ({
   const [showSignStatus, setShowSignStatus] = useState(false)
+  const [restricted, setRestricted] = useState(false)
+
+  useEffect(() => {
```
imo `useMemo` is the more fitting hook in this case. `restricted` is basically a derivation of `walletBlockList` and `creator`.

xat commented on src/pages/objkt-display/index.js:
```diff
@@ -19,10 +20,10 @@ import { Page, Container, Padding } from '@components/layout'
 import styles from './styles.module.scss'
 import './style.css'
-import { Transfer } from '@components/collab/show/Transfer'
+import useSettings from 'hooks/use-settings'
```
`hooks/use-settings` => `@hooks/use-settings`

melMass replied:
Yep, I have a few of them, that's vscode auto import, strangely it works 😅

xat commented on src/pages/objkt-display/index.js:
```diff
@@ -78,10 +81,14 @@ export const ObjktDisplay = () => {
     await context.setAccount()
 
-    if (objkt.ban.includes(objkt.creator.address)) {
+    if (walletBlockList.includes(objkt.creator.address)) {
```
wondering if it wouldn't be better to make `walletBlockList` a hashmap to make these lookups less expensive.

melMass replied:
Like {address: restrictedStatus} ? I can look into that 👁️

xat replied:
jep, like that. this hashmap could be created in the `useSettings` hook.

xat commented on src/pages/tags/index.js:
```diff
@@ -8,7 +8,7 @@ import { renderMediaType } from '@components/media-types'
 import { PATH } from '@constants'
 import styles from './styles.module.scss'
-import { getWalletBlockList } from '../../constants'
+import useSettings from 'hooks/use-settings'
```
`@hooks`

Zir0h commented:
2 small issues:
- pdf files are no longer loading, I used to be able to flip the pages, now it hangs on "Loading PDF..." https://teia-ui-pr-187.onrender.com/objkt/774424
- video identicons don't load, it just says "identicon" https://teia-ui-pr-187.onrender.com/jjjjjohn

xat commented:
> * video identicons don't load, it just says "identicon"

these issues might also be related to the refactorings.

melMass committed:
fix: ⚡️ video identicons

melMass committed:
fix: 🐛 xat suggestions 🥰

melMass committed:
fix: ⚡️ js typing

melMass committed:
fix: ⚡️ pdf

- use the builtin SW (still need testing)
- embed the dist files from react-pdf (cmaps, fonts)
- better loading logic.

melMass committed:
fix: 🎨 embed the pdf SW

I opened an issue https://github.com/wojtekmaj/react-pdf/issues/1048
As I'm not sure how it's handled.

melMass committed:
fix: ⚡️ stop loading pdfs from the feeds

melMass committed:
fix: 🎨 use raw pdf from CDN

melMass commented:
> these issues might also be related to the refactorings.

Actually the logic for falling back to video was only used by media-types (image) component, for now I duplicated the logic for identicons. 🙈

melMass committed:
fix: ⚡️ woff2 is supported by all browsers

The ttf version was 5mb...

melMass committed:
Merge branch 'develop' into mel/feat/tc-thumnails

melMass committed:
fix: ⚡️ use cache.teia.art for generatives

melMass committed:
fix: ⚡️ use raw for display_uri?

melMass committed:
fix: ⚡️ use a map for the ban list
